source/Calamari.Aws/Deployment/Conventions/UploadAwsS3Convention.cs

```diff
@@ -246,9 +246,16 @@ public Task<S3UploadResult> UploadUsingPackage(Func<AmazonS3Client> clientFactor
             Guard.NotNull(deployment, "Deployment may not be null");
             Guard.NotNull(options, "Package options may not be null");
             Guard.NotNull(clientFactory, "Client factory must not be null");
-
+            
+            var id = deployment.Variables.Get(SpecialVariables.Packages.PackageId(null));
+            var version = deployment.Variables.Get(SpecialVariables.Packages.PackageVersion(null));
```

Comment:
We really call `SpecialVariables.Packages.PackageVersion()` with `null` ?!? Is that the convention?

Reply:
Don't get me started :p